### ISSUE TITLE
There is a memory leak defect in line 626 of the file /openssl/crypto/cms/cms_smime.c.

### ISSUE THREAD

**Issue by LuMingYinDetect** [openssl/openssl#24336]
1.In line 561 of the file /openssl/crypto/cms/cms_smime.c, a pointer variable named rct_si is defined. This variable is allocated a block of dynamic memory through the function CMS_add1_signer on line 588. When the if statement on line 589 returns false, it indicates that the dynamic memory allocation for rct_si was successful. When the if statement on line 595 evaluates to true, the program jumps to the err label on line 596 and returns on line 626. During this process, the dynamic memory area pointed to by rct_si is neither used as in line 605 nor released, thus constituting a memory leak defect. See the illustration below:
https://github.com/LuMingYinDetect/openssl_defects/blob/main/openssl_19.png

2.The function CMS_add1_signer mentioned above is responsible for allocating a block of dynamic memory and returning it. Specifically, in line 339, a pointer variable named si is defined. This variable is allocated a block of dynamic memory through the function M_ASN1_new_of on line 351 and returned on line 526. See the illustration below:
https://github.com/LuMingYinDetect/openssl_defects/blob/main/openssl_20.png

**Comment by bernd-edlinger** [comment]
No, this is not a bug.  The value is stored by `CMS_add1_signer()` in the `cms` object:
https://github.com/openssl/openssl/blob/1c4f9684696bad3a602b388a414f2051f0365b3d/crypto/cms/cms_sd.c#L525-L530
